### PR TITLE
Add utils to check for lint and error classes

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
A small step towards lint clean-up that will eventually be needed by https://github.com/r-lib/lintr/pull/2676.

### CODE REVIEW

File: R/get_source_expressions.R [hunk 1]
```diff
@@ -502,7 +502,7 @@ get_source_expression <- function(source_expression, error = identity) {
     error = error
   )
 
-  if (inherits(parsed_content, c("error", "lint"))) {
+  if (is_error(parsed_content) || is_lint(parsed_content)) {
```
Comment (AshesITR) [inline R/get_source_expressions.R:505]: This should be `&&` IINM.

File: R/get_source_expressions.R [hunk 2]
```diff
@@ -513,7 +513,7 @@ get_source_expression <- function(source_expression, error = identity) {
...
```
Comment (AshesITR) [inline R/get_source_expressions.R:516]: ditto

File: R/get_source_expressions.R (hunk 1 above)
Comment (IndrajeetPatil) [reply R/get_source_expressions.R:505]: The original code is conveying that either of those classes is fine, or at least that's what `inherits()` entails:

``` r
x <- tibble::tibble()
y <- data.frame()

class(x)
#> [1] "tbl_df"     "tbl"        "data.frame"
class(y)
#> [1] "data.frame"

inherits(x, c("tbl_df", "data.frame"))
#> [1] TRUE
inherits(y, c("tbl_df", "data.frame"))
#> [1] TRUE
```

<sup>Created on 2024-11-03 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

Comment (AshesITR) [reply R/get_source_expressions.R:505]: Ah, thanks for clarifying.

**Review by AshesITR — APPROVED** [review]

**Pull request merged**